src/adapters/storage.ts

```diff
@@ -74,33 +74,88 @@ export function removeCustomToken(token: Token): Token[] {
 }
 
 // Pending txs
-export function getPendingTxs(env: Env): PendingTx[] {
+export function getAccountPendingTxs(account: string, env: Env): PendingTx[] {
+  const pendingTxs = getPendingTxs(env);
+  return pendingTxs[account] || [];
+}
+
+function getPendingTxs(env: Env): Record<string, PendingTx[]> {
   return getStorageByKey({
     key: constants.PENDING_TXS_KEY,
-    defaultValue: [],
-    parser: z.array(z.unknown()).transform((unknownList) =>
-      unknownList.reduce((acc: PendingTx[], unknown: unknown) => {
-        const pendingTx = pendingTxParser(env).safeParse(unknown);
-        return pendingTx.success ? [...acc, pendingTx.data] : acc;
-      }, [])
+    defaultValue: {},
+    parser: z.record(z.array(z.unknown())).transform((unknownRecord) =>
+      Object.entries(unknownRecord).reduce(
+        (
+          acc: Record<string, PendingTx[]>,
+          [account, unknownList]: [string, unknown[]]
+        ): Record<string, PendingTx[]> => {
+          const pendings = unknownList.reduce(
+            (accArray: PendingTx[], unknown: unknown): PendingTx[] => {
+              const pendingTx = pendingTxParser(env).safeParse(unknown);
+              return pendingTx.success ? [...accArray, pendingTx.data] : accArray;
+            },
+            []
+          );
+          const pendingTxs: Record<string, PendingTx[]> = {
+            ...acc,
+            [account]: pendings,
+          };
+          return pendingTxs;
+        },
+        {}
+      )
     ),
   });
 }
 
-export function setPendingTxs(pendingTxs: PendingTx[]): PendingTx[] {
+export function setPendingTxs(
+  pendingTxs: Record<string, PendingTx[]>
+): Record<string, PendingTx[]> {
   setStorageByKey({
     key: constants.PENDING_TXS_KEY,
-    value: pendingTxs.map(serializePendingTx),
+    value: Object.entries(pendingTxs).reduce(
+      (acc, [key, value]) => ({
+        ...acc,
+        [key]: value.map(serializePendingTx),
+      }),
+      {}
+    ),
   });
+
   return pendingTxs;
 }
 
-export function addPendingTx(env: Env, pendingTx: PendingTx): PendingTx[] {
-  return setPendingTxs([...getPendingTxs(env), pendingTx]);
+export function addPendingTx(account: string, env: Env, pendingTx: PendingTx): PendingTx[] {
+  const pendingTxs = getPendingTxs(env);
+  const pendings: PendingTx[] = pendingTxs[account] || [];
```

Comment:
I think that we can just do:

```
const pendingTxs = getAccountPendingTxs(account, env);
```

Reply:
If we have a function called `getAccountPendingTxs`, this should be called `addAccountPendingTx` as it receives an `account` as a param as well. Same for the `removePendingTx`.

Reply:
Regarding this:

> const accountPendingTxs = getAccountPendingTxs(account, env);

We also neer to call `getPendingTxs(env)` since we need the record, so I prefer the current solution.